src/github/handlers/help-command.ts

```diff
@@ -0,0 +1,78 @@
+import { getConfig } from "../utils/config";
+import { GithubPlugin, isGithubPlugin } from "../types/plugin-configuration";
+import { GitHubContext } from "../github-context";
+
+export interface Command {
+  command: string;
+  description: string;
+  example: string;
+}
+
+export interface Manifest {
+  name: string;
+  description: string;
+  commands: Command[];
+}
+
+export async function postHelpCommand(context: GitHubContext<"issue_comment.created">) {
+  const comments = [
+    "### Available Commands\n\n",
+    "| Command | Description | Example |",
+    "|---|---|---|",
+    "| `/help` | List all available commands. | `/help` |",
+  ];
+  const commands: string[] = [];
+  const configuration = await getConfig(context);
+  for (const pluginArray of Object.values(configuration.plugins)) {
+    for (const pluginElement of pluginArray) {
+      const { plugin } = pluginElement.uses[0];
+      const manifest = await (isGithubPlugin(plugin) ? fetchActionManifest(context, plugin) : fetchWorkerManifest(plugin));
+      if (manifest?.commands) {
+        for (const command of manifest.commands) {
+          commands.push(`| \`${getContent(command.command)}\` | ${getContent(command.description)} | \`${getContent(command.example)}\` |`);
+        }
+      }
+    }
+  }
+  await context.octokit.issues.createComment({
+    body: comments.concat(commands.sort()).join("\n"),
+    issue_number: context.payload.issue.number,
+    owner: context.payload.repository.owner.login,
+    repo: context.payload.repository.name,
+  });
+}
+
+/**
+ * Ensures that passed content does not break MD display within the table.
+ */
+function getContent(content: string | undefined) {
+  return content ? content.replace("|", "\\|") : "-";
+}
+
+async function fetchActionManifest(context: GitHubContext<"issue_comment.created">, { owner, repo }: GithubPlugin): Promise<Manifest | null> {
+  try {
+    const { data } = await context.octokit.repos.getContent({
+      owner,
+      repo,
+      path: "manifest.json",
+    });
+    if ("content" in data) {
+      const content = Buffer.from(data.content, "base64").toString();
+      return JSON.parse(content);
+    }
+  } catch (e) {
+    console.warn(`Could not find a manifest for ${owner}/${repo}`);
```

Comment:
We should force plugin developers to include a manifest. Perhaps a plugin can "fail to install" without it? 

What do you think?

Lastly, it would be very interesting to have a schema defined for text editors of our manifests, a bit like how the linters do it.

Reply:
@0x4007 I will add it in the plugin template as well. Currently if there is no manifest I just ignore the repo as you might have seen, the reason being that plugins typically don't really need it if they have no command. Not sure if it is better to enforce it or not.

I agree for the linter. Actually because it is currently named `manifest.json` my IDE auto-completes as if it was for a website. I don't know if we could that format for ourselves as well
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json

Doing a custom linter would be tedious. However I wanna add a manifest content check, the same way we do for the configuration.

Reply:
> @0x4007 I will add it in the plugin template as well. Currently if there is no manifest I just ignore the repo as you might have seen, the reason being that plugins typically don't really need it if they have no command. Not sure if it is better to enforce it or not.
> 
> 
> 
> I agree for the linter. Actually because it is currently named `manifest.json` my IDE auto-completes as if it was for a website. I don't know if we could that format for ourselves as well
> 
> https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json
> 
> 
> 
> Doing a custom linter would be tedious. However I wanna add a manifest content check, the same way we do for the configuration.

Another idea is to dynamically generate it with a small script inside of the plugin template? 

Perhaps if manifest.json is requested but doesn't exist, it can serve up one based on details described in package.json?

--- 

I think it's a good idea to follow the existing web standards for the manifest schema.

Reply:
The standard web schema has a `commands` section but it doesn't exactly suit our needs. We could generate something from the package.json but I don't think it would be able to describe commands with examples, since theses are not things we put in it. Just having the manifest in the template should help, I think.